libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts

```diff
@@ -84,24 +72,19 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> {
         this.dialogs.telemetryClient = client;
     }
 
-    public addRule(rule: OnCondition): this {
-        this.triggers.push(rule);
-        return this;
-    }
-
     protected ensureDependenciesInstalled(): void {
         if (this.installedDependencies) {
             return;
         }
         this.installedDependencies = true;
 
-        // Install any actions
-        this.actions.forEach((action) => this.dialogs.add(action));
-
         // Install each trigger actions
-        this.triggers.forEach((trigger) => {
-            trigger.actions.forEach((action) => this.dialogs.add(action));
-        });
+        for (let i = 0; i < this.triggers.length; i++) {
+            const trigger = this.triggers[i];
+            if (typeof ((trigger as any) as DialogDependencies).getDependencies == 'function') {
+                ((trigger as any) as DialogDependencies).getDependencies().forEach((child) => this.dialogs.add(child));
+            }
```

Comment:
You don't need to do this. When a dialog is added it will recursively have it's dependencies added. The C# implementation is wrong and should be changed. The old code was correct.